graphics/opencv/Portfile

```diff
@@ -277,6 +277,12 @@ variant opencl description {Enable OpenCL support} {
                             -DWITH_OPENCL=ON
 }
 
+variant openmp description {Include OpenMP support (untested)} {
+    configure.args-replace \
+                            -DWITH_OPENMP=OFF \
+                            -DWITH_OPENMP=ON
```

Comment:
Note: Unlike most other variants, `openmp` doesn't involve additional dependencies.

Reply:
I didn't merge this because you're not specifying an openmp compatible compiler using MacPorts' facilities for this:

```compiler.openmp_version```

see this page:

<https://trac.macports.org/wiki/CompilerSelection>

Reply:
and also because you say it is "untested" and I just don't see how you can really enable that without testing whether it works, to be honest...

So -- seems to me you have a bit more work to do here before this is ready for prime-time.

Reply:
> and also because you say it is "untested" and I just don't see how you can really enable that without testing whether it works, to be honest...

Well, this variant isn't untested - I've certainly installed with it enabled - but it's essentially a plaseholder at this point: OpenMP isn't found during the configure phase, so OpenCV is built without it.

It would probably be more accurate to change the variant description to "Not yet implemented." But my priority was to bring both ports up to parity first, and tackle this type of cleanup during phase five. (That'll be when all OpenCV-related ports are converted to subports, duplicated logic removed, etc.)

Reply:
Updated variant descriptions to state "Not Yet Implemented," rather than untested.

Reply:
if you add this in the variant, with the proper setting, then it will probably be found during the configure phase, and actually might work:
```
compiler.openmp_version
```

Reply:
> if you add this in the variant, with the proper setting, then it will probably be found during the configure phase, and actually might work:
> 
> ```
> compiler.openmp_version
> ```

Hmmmmm. Well, given that it'll need to tested against macOS 10.8 through 10.15 - plus other unforeseen issues, like incorrect compiler selection on certain macOS/Xcode combinations (per [Base PR 214](https://github.com/macports/macports-base/pull/214), which won't be in-play until the next MacPorts release... I'm really not feeling enthused about going through that right now.  :-O

Reply:
Updated description for OpenCV reconciliation tracking ticket, specifically calling out the OpenMP variant:

[opencv/opencv4: eliminate portfile duplication, via subport](https://trac.macports.org/ticket/62011)

Reply:
Ken, I don't suppose you'd be willing to give me a pass on this one, given that it's going to be tackled in short order?

Reply:
Okay, since I have to fire up my VMs for other testing anyway, I'll give your suggestion a try, and report back.

Reply:
Ken, is specifying `compiler.openmp_version` enough? Or do I also need to set `compiler.mpi`? (For the latter, I'd probably select `openmpi`... unless there are any other gotchas I need to be aware of...?)

Reply:
When `compiler.mpi` is specified too, that switches the compiler to `openmpi-clang90`. Which is arguably a good thing, assuming that's what's needed for true OpenMP support...?

However, there's a problem: I haven't installed `openmpi-clang90` anywhere yet. And given that it requires a source build, it takes a LONG time. Even on my physical macOS instances, with eight real CPU cores, it takes forever to build. On my VMs - which are limited to four vCPUs (I don't have Parallels Professional) - it's horrific.

In any case, let me know if `compiler.mpi` is required. If not, the changes are still feasible. But if it is required, I'm going to punt on this for now.

Ken, let me know your thoughts.

Reply:
I'm currently testing with `compiler.openmp_version` set to 4.0, and that does seem to do the trick. However, running through my entire suite of regression tests - which would involve rebuilding downstream ports, and testing their functionality - just isn't feasible right now.

So, if we want to add this change - with the caveat that it's not well-tested, beyond building and simple binary testing of opencv/opencv4 - we can do it.

How shall we proceed?

Reply:
Pushed the changes, to save time. We can always back them out if that's not the approach we want to take.